### DISCUSSION TITLE
Gitlab integration not working on Windows

### DISCUSSION THREAD

**Discussion by Mcluky** [codemagic-ci-cd/codemagic-docs#1774] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Setup Gitlab integration in Codemagic Teams sections
2. Create a pipeline that uses the Windows build agent
3. Run pipeline

### Expected results

The pipeline is expected to check out the repository and run further pipeline steps (like on Mac and Linux where this works)

### Actual results

The pipeline fails to check out the repository.

There is a workaround by using ssh based authentication but this is bad because you lose Gitlab integration features like build status report on the pull request and so.  

### Build id (optional)

648ac1ca6f1357c56fd67f2c

**Comment by icarusdust** [comment]
Hi @Mcluky, please add your repo via its SSH url using the other option when adding a new application. Let me know if it still throws the same error afterwards

**Reply by Mcluky** [reply]
Hi @icarusdust as I mentioned in my original report

        There is a workaround by using ssh based authentication but this is bad because you lose Gitlab integration features like build status report on the pull request and so.

SSH checkout does work **but** it doesn't report the build status back to Gitlab. My team heavily relies on this with merge request checks before merging. With SSH this doesn't work anymore.

**Reply by icarusdust** [reply]
That's right @Mcluky , sorry missed that part of your message. 

I have started discussing it internally with the team and will keep you updated as soon as I have updates.

Though I must mention one important thing that your team is on a paid plan which means that you should have access to the private chat widget in the UI. Please use that next time for quicker responses.

**Reply by Mcluky** [reply]
That's true. We are on the paid plan and I have already reached out to your support on the private chat. They even initially gave me the hint with the SSH-based authentication workaround but when I mentioned that this doesn't really work for us they refused to further help us since a workaround has been provided. 

I've created this issue here on GitHub mainly for troubleshooting purposes for the community since I have not been able to find this issue reported anywhere and maybe someone else has this problem as well. And perhaps it might gain some traction here.

**Reply by icarusdust** [reply]
@Mcluky I totally understand your point and sorry that you did not get any further assistance with this issue in the private chat. Please note that I have already raised it to the team and we will revisit the topic of enabling PR checks depending on your use case. Let me not give you an ETA right now, though it is in our plan.

**Reply by Mcluky** [reply]
@icarusdust Is there a way to subscribe to new releases/fixes you've made? We have changed all our pipelines to ssh based auth and disabled our merge requests checks however we want to go back to that asap. How would I know when you fixed the issue without having to try it over and over again? An ETA would be really useful for us.

**Reply by icarusdust** [reply]
@Mcluky, I cannot give you an ETA at the moment, but I can keep you posted as soon as there are updates.

**Comment by icarusdust** [comment]
I will close this thread for now until further updates. You can re-open it by replying to it.
